src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -285,6 +323,93 @@ private void validateSupportedTypes(Schema.Field field) {
     }
   }
 
+  private void validatePartitionProperties() {
+
+    Table sourceTable = BigQueryUtil.getBigQueryTable(config.getDatasetProject(), config.getDataset(),
+                                                      config.getTable(), config.getServiceAccountFilePath());
+    TimePartitioning timePartitioning = ((StandardTableDefinition) Objects.requireNonNull(sourceTable).getDefinition())
+      .getTimePartitioning();
+    if (timePartitioning == null) {
+      return;
+    }
+    String partitionFromDate = config.getPartitionFrom();
+    String partitionToDate = config.getPartitionTo();
+
+    if (partitionFromDate == null && partitionToDate == null) {
+      return;
+    }
+    LocalDate fromDate = null;
+    if (partitionFromDate != null) {
+      try {
+        fromDate = LocalDate.parse(partitionFromDate);
+      } catch (DateTimeException ex) {
+        throw new InvalidConfigPropertyException("Invalid Partition From Date format. Partition From Date should be " +
+                                                   "in format yyyy-MM-dd", ex, "Partition From Date");
```

Comment:
Last parameter in the exception should be property API name. Meaning "someProperty" instead of "someProperty". As far a I see from examples. Please mark the properties in config with @Name and use the same constant in the exceptions.

Reply:
Done.